pkgs/applications/editors/neovim/wrapper.nix

```
@@ -40,22 +47,59 @@ let
     # lua code to put into the generated init.lua file
     , luaRcContent ? ""
     # entry to load in packpath
-    , packpathDirs
+    , packpathDirs ? null # not used anymore
+    , plugins ? []
     , ...
-  }:
+  }@attrs:
   assert withPython2 -> throw "Python2 support has been removed from the neovim wrapper, please remove withPython2 and python2Env.";
 
+  # assert packpathDirs -> throw "Packpathdirs removed";
```

```suggestion
  assert lib.assertMsg (attrs ? packpathDirs) "packpathDirs removed";
```

I believe best practice though would be to remove the `...` so that only supported args can be passed? Unless the args truly can be anything and will be passed on to another function.

---

funny I am of the opposite opinion ^^' it lets the package be forward and backward compatible (you can mix different versions of nixpkgs) and add warnings such as this one.

---

> I believe best practice though would be to remove the ... so that only supported args can be passed?

I believe I was advised this when contributing to `/lib`.

IIRC the argument was something along the lines of: it's better for the function signature to _not_ be forward/backward compatible in ways that the function _itself_ is not.

e.g. if someone uses a feature of the function that is supported by a _newer_ nixpkgs than they have, they would not know that the feature has no effect.

Backwards compatibility can still be achieved by keeping stub `foo ? null,` parameters in the signature, but forward compatibility is more difficult because we don't know what will be added in the future.

Either way, this PR isn't _adding_ the `...` so this isn't feedback on the PR itself, just commenting on the existing code 🙃